js_modules/dagster-ui/packages/ui-core/src/workspace/WorkspaceContext.tsx

```diff
@@ -59,170 +63,151 @@ export const WorkspaceContext = React.createContext<WorkspaceState>(
 
 export const HIDDEN_REPO_KEYS = 'dagster.hidden-repo-keys';
 
-export const ROOT_WORKSPACE_QUERY = gql`
-  query RootWorkspaceQuery {
-    workspaceOrError {
-      ... on Workspace {
-        id
-        locationEntries {
-          id
-          ...WorkspaceLocationNode
-        }
-      }
-      ...PythonErrorFragment
-    }
-  }
+export const WorkspaceProvider = ({children}: {children: React.ReactNode}) => {
+  const {localCacheIdPrefix} = useContext(AppContext);
+  const codeLocationStatusQueryResult = useIndexedDBCachedQuery<
+    CodeLocationStatusQuery,
+    CodeLocationStatusQueryVariables
+  >({
+    query: CODE_LOCATION_STATUS_QUERY,
+    version: CODE_LOCATION_STATUS_QUERY_VERSION,
+    key: `${localCacheIdPrefix}/${CODE_LOCATION_STATUS_QUERY_KEY}`,
```

Comment:
can we push `localCacheIdPrefix` handling down in to `CacheManager`?

Reply:
Hmm, Yeah I think I could make it more seamless if I track it with a global instead of trapping it in React Context (currently it can only be pulled by Hooks/Components)

Reply:
I'd like to do this in a follow up PR though